include/drivers/dac.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2020 Libre Solar Technologies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief DAC public API header file.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_DAC_H_
+#define ZEPHYR_INCLUDE_DRIVERS_DAC_H_
+
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief DAC driver APIs
+ * @defgroup dac_interface DAC driver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/** @brief DAC references. */
+enum dac_reference {
+	DAC_REF_DEFAULT,   /**< Default, reference cannot be selected. */
+	DAC_REF_VDD_1,     /**< VDD. */
+	DAC_REF_VDD_1_2,   /**< VDD/2. */
+	DAC_REF_VDD_1_3,   /**< VDD/3. */
+	DAC_REF_VDD_1_4,   /**< VDD/4. */
+	DAC_REF_INTERNAL,  /**< Internal. */
+	DAC_REF_EXTERNAL0, /**< External, input 0. */
+	DAC_REF_EXTERNAL1, /**< External, input 1. */
+};
+
+/**
+ * @struct dac_channel_cfg
+ * @brief Structure for specifying the configuration of a DAC channel.
+ *
+ * @param channel_id Channel identifier of the DAC that should be configured.
+ * @param resolution Desired resolution of the DAC (depends on device
+ *                   capabilities).
+ * @param reference Reference selection.
+ * @param buffered Enable buffered output if available in underlying hardware.
+ */
+struct dac_channel_cfg {
+	u8_t channel_id;
+	u8_t resolution;
+	enum dac_reference reference;
+	bool buffered;
+};
+
+/**
+ * @cond INTERNAL_HIDDEN
+ *
+ * For internal use only, skip these in public documentation.
+ */
+
+/*
+ * Type definition of DAC API function for configuring a channel.
+ * See dac_channel_setup() for argument descriptions.
+ */
+typedef int (*dac_api_channel_setup)(struct device *dev,
+				const struct dac_channel_cfg *channel_cfg);
+
+/*
+ * Type definition of DAC API function for setting a write request.
+ * See dac_write_value() for argument descriptions.
+ */
+typedef int (*dac_api_write_value)(struct device *dev,
+				    u8_t channel, u32_t value);
+
+/*
+ * DAC driver API
+ *
+ * This is the mandatory API any DAC driver needs to expose.
+ */
+__subsystem struct dac_driver_api {
+	dac_api_channel_setup channel_setup;
+	dac_api_write_value   write_value;
+	u16_t ref_internal;	/* mV */
```

Comment:
Repeating the comment that didn't get saved: This doesn't handle all cases, as described in https://github.com/zephyrproject-rtos/zephyr/pull/24292#issuecomment-612605361 so it might be worth not following ADC in this case.